src/System.Private.CoreLib/src/System/Environment.CoreCLR.cs

```diff
@@ -0,0 +1,248 @@
+            {
+                if (pStrings != null)
+                {
+                    Win32Native.FreeEnvironmentStrings(pStrings);
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern string[] GetCommandLineArgsNative();
+
+        public static string[] GetCommandLineArgs()
+        {
+            // There are multiple entry points to a hosted app. The host could
+            // use ::ExecuteAssembly() or ::CreateDelegate option:
+            //
+            // ::ExecuteAssembly() -> In this particular case, the runtime invokes the main 
+            // method based on the arguments set by the host, and we return those arguments
+            //
+            // ::CreateDelegate() -> In this particular case, the host is asked to create a 
+            // delegate based on the appDomain, assembly and methodDesc passed to it.
+            // which the caller uses to invoke the method. In this particular case we do not have
+            // any information on what arguments would be passed to the delegate.
+            // So our best bet is to simply use the commandLine that was used to invoke the process.
+            // in case it is present.
+
+            return s_commandLineArgs != null ?
+                (string[])s_commandLineArgs.Clone() :
+                GetCommandLineArgsNative();
+        }
+
+        public static extern bool HasShutdownStarted
+        {
+            [MethodImpl(MethodImplOptions.InternalCall)]
+            get;
+        }
+
+        public static int ProcessorCount => GetProcessorCount();
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern int GetProcessorCount();
+
+        private static void SetEnvironmentVariableCore(string variable, string value)
+        {
+            if (!Win32Native.SetEnvironmentVariable(variable, value))
+            {
+                int errorCode = Marshal.GetLastWin32Error();
+                switch (errorCode)
+                {
+                    case Interop.Errors.ERROR_ENVVAR_NOT_FOUND:
+                        // Allow user to try to clear a environment variable
+                        return;
+
+                    case Interop.Errors.ERROR_FILENAME_EXCED_RANGE:
+                        // The error message from Win32 is "The filename or extension is too long",
+                        // which is not accurate.
+                        throw new ArgumentException(SR.Format(SR.Argument_LongEnvVarValue));
+
+                    case Interop.Errors.ERROR_NOT_ENOUGH_MEMORY:
+                    case Interop.Errors.ERROR_NO_SYSTEM_RESOURCES:
+                        throw new OutOfMemoryException(Interop.Kernel32.GetMessage(errorCode));
+
+                    default:
+                        throw new ArgumentException(Interop.Kernel32.GetMessage(errorCode));
+                }
+            }
+        }
+
+        public static string StackTrace
```

Comment:
Looks likes something what could be shared

Reply:
It has a very different implementation on CoreRT:
https://github.com/dotnet/corert/blob/715f00085a3cf34f493e019c5fd90224de078bf9/src/System.Private.CoreLib/src/Internal/Runtime/Augments/EnvironmentAugments.cs#L124-L149